tests/test_transpiler_router.py

```diff
@@ -331,3 +331,36 @@ def test_sabre_memory_map():
     router._memory_map = [[1, 0, 2, 3, 4]]
     value = router._compute_cost((0, 1))
     assert value == float("inf")
+
+
+def test_sabre_intermediate_measurements():
+    measurement = gates.M(0)
+    circ = Circuit(3, density_matrix=True)
+    circ.add(gates.H(0))
+    circ.add(measurement)
+    circ.add(gates.CNOT(0, 1))
+    connectivity = nx.Graph()
+    connectivity.add_nodes_from([0, 1, 2])
+    connectivity.add_edges_from([(0, 1), (1, 2)])
+    router = Sabre(connectivity=connectivity)
```

Comment:
But here Sabre is doing nothing right? The CNOT can be executed without any SWAP. I would test the scenery where the measurement is actually moved by the routing.